src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -304,8 +309,6 @@ def _look_ahead_match(cls, segments, matchers, parse_context):
 
         # Do some type munging
         matchers = list(matchers)
```

Comment:
A bit off topic for the PR, but a possible performance improvement: I looked through the function, and I see no reason to make a new list here. Could we remove this copying? It's likely to have some impact on performance.